app/notification_email.go

```diff
@@ -283,6 +287,38 @@ func getFormattedPostTime(user *model.User, post *model.Post, useMilitaryTime bo
 	}
 }
 
+func (a *App) generateHyperlinkForChannels(postMessage, teamName, teamURL string) string {
+
+	team, err := a.GetTeamByName(teamName)
+	if err != nil {
+		mlog.Error(fmt.Sprintf("Encountered error while looking up team by name for %s with error: %s", teamName, err.Error()))
+		return postMessage
+	}
+
+	channelNames := model.ChannelMentions(postMessage)
+	if len(channelNames) == 0 {
+		mlog.Debug(fmt.Sprintf("No Channel Mentions"))
+		return postMessage
+	}
+
+	channels, err := a.GetChannelsByNames(channelNames, team.Id)
+	if err != nil {
+		mlog.Error(fmt.Sprintf("Encountered error while getting channels with error: %s", err.Error()))
```

Comment:
Can you change this to structured parameters?
```suggestion
		mlog.Error("Encountered error while getting channels", mlog.Err(err))
```